index.d.ts

```diff
@@ -133,6 +133,38 @@ type ReaderRevenuePosition =
     | 'ampHeader'
     | 'ampFooter';
 
+interface CampaignsType {
+    id: string;
+    name: string;
+    rules: [];
+    priority: number;
+    activeFrom: number;
+    displayOnSensitive: boolean;
+    fields: {
+        formId: number;
+        callout: string;
+        _type: string;
+        description: string;
+        tagName: string;
+        formFields: CampaignsFeildType[];
+    };
```

Comment:
Is there a point to having this object  property? Could you just not have it?

Reply:
I agree, the structure is strange, `fields` seems to always be the same it only `formFields` that changes dynamically, so i think it makes sense to keep it as is

Reply:
I think it's a mistake to build our types based on the data we receive. We should do the opposite, and build the types based on what we use